compiler/hash-types/src/new/lits.rs

```diff
@@ -51,3 +53,48 @@ pub enum LitTerm {
     Char(CharLit),
     Float(FloatLit),
 }
+
+impl Display for IntLit {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}{}", self.underlying.value, self.underlying.kind)
+    }
+}
+
+impl Display for StrLit {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.underlying.data)
+    }
+}
+
+impl Display for CharLit {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.underlying.data)
```

Comment:
should use debug printing due to special characters like `❤` not being printed as the value of the character, but `\u{...}` in the event that it does not have a defined glyph.